sdk/synapse/Azure.Analytics.Synapse.Spark/src/Customization/SparkBatchOperation.cs

```diff
@@ -146,14 +162,23 @@ private static bool IsJobComplete(string jobState, string livyState)
 
             switch (livyState)
             {
-                case "starting":
                 case "error":
                 case "dead":
                 case "success":
                 case "killed":
                     return true;
             }
 
+            if (!shouldWaitForJobExecution)
```

Comment:
When adding a functional change like this, it is best to add a corresponding test validating the functionality for two reasons:
1. It provides a sample that illustrates the intended use case
2. It prevents regressions in the case that someone else make changes elsewhere without understanding this requirement.

Reply:
Good suggestions. I will add test cases.